android/src/main/kotlin/top/kikt/imagescanner/core/utils/Android30DbUtils.kt

```diff
@@ -376,7 +376,14 @@ object Android30DbUtils : IDBUtils {
     val cr = context.contentResolver
     val timestamp = System.currentTimeMillis() / 1000
 
-    val typeFromStream = URLConnection.guessContentTypeFromStream(inputStream)
+    var typeFromStream: String;
+    if (title.contains(".")){
+      //title contains file extension, form mimeType from it
+      typeFromStream = "image/${File(title).extension}"
+    } else {
+      typeFromStream = URLConnection.guessContentTypeFromStream(inputStream)
+              ?: "image/${File(title).extension}"
```

Comment:
Could `.extension` promote the correct type here since title doesn't contain `.` ?

Reply:
Not really, but I left it since it was there.... ^^
I think if no extension is specified in the title and `guessContentTypeFromStream` returns null, the mimeType should just be set to `image/*`

Reply:
> the mimeType should just be set to `image/*`

Sure it's definitely better :) .